lib/wallaroo/core/windows/panes_range_windows.pony

```diff
@@ -119,16 +123,41 @@ class _PanesSlidingWindows[In: Any val, Out: Any val, Acc: State ref] is
           _expand_windows(event_ts, new_end_ts)?
           new_earliest_ts = _earliest_ts()?
         end
+        is_late_data = _apply_input(input, event_ts, new_earliest_ts)
+      end
 
-        _apply_input(input, event_ts, new_earliest_ts)
+      if is_late_data then
+        match _late_data_policy
+        | LateDataPolicy.drop() =>
+          ifdef debug then
+            @printf[I32]("Event ts %s is earlier than earliest window %s. Ignoring\n".cstring(), event_ts.string().cstring(), earliest_ts.string().cstring())
+          end
+        | LateDataPolicy.fire_per_message() =>
+          let acc = _agg.initial_accumulator()
+          _agg.update(input, acc)
+          // We are currently using the event timestamp of the late message
+          // as the window end timestamp.
+          match _agg.output(_key, event_ts, acc)
```

Comment:
@jtfmumm I think we should add a test to make sure that this is indeed the timestamp of the output window. See: https://github.com/WallarooLabs/wallaroo/blob/master/lib/wallaroo/core/windows/_windows_test.pony#L97 for an example of how we did it with the regular range window tests.

Reply:
Added test and related fix.